src/services/reviews.js

```diff
@@ -0,0 +1,672 @@
+				const validateNextLevel = await this.validateNextLevel(userRoles, userOrgId, resourceType, nextStage)
+				if (validateNextLevel.statusCode !== httpStatusCode.ok) {
+					return validateNextLevel
+				}
+
+				updateNextLevel = true
+			}
+
+			//create a new entry in reviews and review_resources table
+			const reviewData = {
+				resource_id: resourceId,
+				reviewer_id: userId,
+				status: common.REVIEW_STATUS_STARTED,
+				organization_id: userOrgId,
+			}
+
+			const newReview = await reviewsQueries.create(reviewData)
+			if (!newReview?.id) {
+				return responses.failureResponse({
+					message: 'FAILED_TO_START_REVIEW',
+					statusCode: httpStatusCode.internal_server_error,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			await reviewResourceQueries.create(_.omit(reviewData, [common.STATUS]))
+
+			// Update resource table data
+			let updateData = {
+				status: common.RESOURCE_STATUS_IN_REVIEW,
+				last_reviewed_on: new Date(),
+				...(updateNextLevel && { next_stage: nextStage + 1 }),
+			}
+
+			await resourceQueries.updateOne({ organization_id: resourceOrgId, id: resourceId }, updateData)
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_CREATED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Validating the Resource for review creation or update
+	 * @method
+	 * @name validateReview
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} userId - logged in user id.
+	 * @param {String} resourceStatus - resource status
+	 * @param {String} orgId - organization Id
+	 * @returns {JSON} - review validation response.
+	 */
+
+	static async validateReview(resourceId, userId, resourceStatus, orgId) {
+		try {
+			//validate resource status
+			const validateResourceStatus = await this.validateResourceStatus(resourceStatus)
+			if (validateResourceStatus.statusCode !== httpStatusCode.ok) {
+				return validateResourceStatus
+			}
+
+			// Validate review existence and fetch review details
+			const reviewValidation = await this.verifyAndFetchReview(userId, resourceId)
+			if (reviewValidation.statusCode !== httpStatusCode.ok) {
+				return reviewValidation
+			}
+
+			const review = reviewValidation.result
+
+			// Check if another reviewer has already started the review
+			if (review.status === common.REVIEW_STATUS_NOT_STARTED) {
+				const activeReviewCheck = this.validateNoActiveReviewByOthers(resourceId, userId, orgId)
+				if (activeReviewCheck.statusCode !== httpStatusCode.ok) {
+					return activeReviewCheck
+				}
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'VALIDATION_PASSED',
+				result: review,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Validating the Resource status
+	 * @method
+	 * @name validateResourceStatus
+	 * @param {String} status - resource status
+	 * @returns {JSON} - return error if any
+	 */
+
+	static async validateResourceStatus(status) {
```

Comment:
@priyanka-TL change function name to isResourceAvailableForReview, should return true/false.